### DISCUSSION TITLE
Nested fragments based on interface generate too many typescript types - out of memory compilator issue

### DISCUSSION THREAD

**Discussion by Marcinthedev** [dotansimha/graphql-code-generator#5270] [General]
Hi, I have a problem with nested types. Codegen generates huge file of types while nesting fragments. It's that big that compilator has problems with them and runs  out of memory.

I have in schema lots of possible types which implement one interface. I am using this interface in  __resolveTypes to get the correct type of data. On the client side I've created matching fragments and I have a use case that they may be nested. Here is an example:
```
fragment BasicFieldProps on BasicField {
  ... on CheckboxField {
    ...CheckboxFieldFragment
  }
  ... on DateField {
    ...DateFieldFragment
  }
... on GroupField {
    ...GroupFieldFragment
  }
}
```
where `BasicField` is 
```
interface BasicField {
  id: ID!
  name: String!
  key: String!
}
```
and all of the others are types that implement this interface.
The issue here is that I want to return BasicField which may be any of them. When it's GroupField it may have nested elements thus I need to return something like this: 
```
fragment BasicField on BasicField {
  ...BasicFieldProps // any field
  metadata {
    ...BasicFieldMetadata
    fieldsGroup {
      ...BasicFieldProps
      metadata {
             ...BasicFieldMetadata
       }
// etc. maximum is 6 levels deep.
    }
  }
}

```
My case is more complex but the solution is the same. Just more nested elements and more properties. 
Is there a way to may shorten typescript somehow? It also generates DateField with the `fieldsGroup` property while it will never have such data. 

If something is unclear I will try to create a better and complex example

**Comment by dotansimha** [comment]
Thank you for reporting this @Marcinthedev . Can you please share a live reproduction of that issue?
